File: LeetCode/Python/0334_Increasing_Triplet_Subsequence.py
# 334. Increasing Triplet Subsequence
# https://leetcode.com/problems/increasing-triplet-subsequence/


class Solution:
    def increasingTriplet(self, nums):
        i = j = float("inf")
        for k in nums:
            if k <= i:
                i = k
            elif k <= j:
                j = k
            else:
                return True

File: LeetCode/JavaScript/0334-Increasing-Triplet-Subsequence.js
// 334. Increasing Triplet Subsequence
// https://leetcode.com/problems/increasing-triplet-subsequence/description/

/**
 * @param {number[]} nums
 * @return {boolean}
 */

var increasingTriplet = function (nums) {
  let first = Infinity;
  let second = Infinity;

  for (let num of nums) {
    if (num <= first) first = num;
    else if (num <= second) second = num;
    else return true;
  }
  return false;
};
